client/lcd/test_helpers.go

```diff
@@ -193,6 +169,20 @@ func (b AddrSeedSlice) Swap(i, j int) {
 	b[j], b[i] = b[i], b[j]
 }
 
+// InitClientHome initialises client home dir.
+func InitClientHome(t *testing.T, dir string) string {
+	var err error
+	if dir == "" {
+		dir, err = ioutil.TempDir("", "lcd_test")
+		require.NoError(t, err)
+	}
+	// TODO: this should be set in NewRestServer
+	// and pass down the CLIContext to achieve
+	// parallelism.
+	viper.Set(cli.HomeFlag, dir)
```

Comment:
Maybe this could return a cleanup function to defer.
Something similar is being done here: https://github.com/cosmos/cosmos-sdk/blob/e1f0767ba8dd96bac05b89e234ec021d1cce582a/server/test_helpers.go#L43-L53

I am adding a temporary directory helper in a future PR.

Reply:
I don't see this as an urgent change to block the PR